pages/index.tsx

```diff
@@ -124,11 +122,11 @@ function index({
   // If project details are present set project details
   // This will be set from getServerSideProps.
 
-  React.useEffect(() => {
-    if (projectDetails) {
-      setprojectDetails(projectDetails);
-    }
-  }, [projectDetails]);
+  // React.useEffect(() => {
+  //   if (projectDetails) {
+  //     setprojectDetails(projectDetails);
+  //   }
+  // }, [projectDetails]);
```

Comment:
We already once had an PR by @Shreyaschorge with a commit removing this code and then after [reviewing](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/305#discussion_r905980648) added again https://github.com/Plant-for-the-Planet-org/planet-donations/pull/305 - can you please check if this code is not really needed any more for some use case?

Reply:
[Here is another PR](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/273) where the code was activated again and [was commented before](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/254). this part of the code seems to have been changed quite often. I think it is about time to understand the implications of it and find a stable solution.

Reply:
Here is the PR changing the code into an React hook: https://github.com/Plant-for-the-Planet-org/planet-donations/pull/248

Reply:
Btw. the API calls made in `getServerSideProps` do not use the same locale as set in the UI and I do not know how one could solve this. I am wondering if we really still need to use `getServerSideProps` for planet-donation or if we want to refactor this application to do the API calls in the browser instead, but that would be a major refactoring and I am unsure about the initial idea of using that server based architecture, but it surely should only load translation independent data.

Reply:
> We already once had an PR by @Shreyaschorge with a commit removing this code and then after [reviewing](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/305#discussion_r905980648) added again #305 - can you please check if this code is not really needed any more for some use case?

sure, I'll look into it

Reply:
> [Here is another PR](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/273) where the code was activated again and [was commented before](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/254). this part of the code seems to have been changed quite often. I think it is about time to understand the implications of it and find a stable solution.

@norbertschuler the way app is being structured right now one possible solution could be
having query param `locale` always appended to the url. This locale will be the clients locale.

But it will create an issue when a user shares a link to another user. If they both have different locale set in the browser. User1's locale will be enforced on another user.

Reply:
I don't think putting the locale in the query string will work. There are too many use cases (including links) where it can get lost.

Reply:
I think disabling overwriting the projects details with the data from the server side API calls should be further investigated. This MR works on first sight. We now have to make sure no use case is forgotten which demanded loading the project details on server side, e.g. different payment method returning to the application with different callback links.

Reply:
I found a use case where this MR does not work: Returning from an SOFORT payment with an URL like http://localhost:3000/?context=don_cp14EbHAvTGEyf8PUc1wkEGf&country=DE&locale=de&method=Sofort&payment_intent=pi_3MDQXWD2OpW2f42N1kQSggxf&payment_intent_client_secret=pi_3MDQXWD2OpW2f42N1kQSggxf_secret_fQgYJvtm1avtENTdyu31Vz45p&redirect_status=pending&tenant=ten_I9TW3ncG should now with to update the project details and do not show the project:

<img width="861" alt="Bildschirm­foto 2022-12-10 um 12 00 37" src="https://user-images.githubusercontent.com/1532418/206851747-7b0f03d2-9c50-43ca-98a2-756bd762abad.png">

Also in this use case the URL contains a locale making the API call on server side loading the project data in German language which also makes the meta data be German on the resulting page instead of being in English language.